### PR TITLE
feat: install and implement React Router V6 (TBC-22)

### PR DESCRIPTION
**Pull request by ohnomalone**
### PR Checklist

Please check for the following requirements:

- [x] I have reviewed this PR's changes at least once
- [ ] The squashed commit message explains exactly what you changed, added or deleted

#### Changes

- Installs React Router v6
- implements router in app
- create login route `/`
- create - create user route `/create-user`
- create Login and Create User modules and basic components

##### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### What is the current behavior?

- [Ticket](https://travelingbookclub.atlassian.net/browse/tbc-22)
- [ ] Another developer ran through the acceptance criteria locally (bugs fixes and features)
- [x] I ran through the acceptance criteria locally (bugs fixes and features)
- [ ] I have added tests for meeting the acceptance criteria (bugs fixes and features)

Screenshot 2023-05-18 at 4 48 33 PM
Screenshot 2023-05-18 at 7 07 19 PM

### CODE REVIEW

File: web-ui/package.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
     "@emotion/styled": "^11.11.0",
     "@mui/material": "^5.13.0",
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "react-router-dom": "^6.11.2"
```
Comment (ohnomalone) [inline web-ui/package.json:18]: installed react router package

File: web-ui/src/App.tsx [hunk 2]
```diff
@@ -1,35 +1,15 @@
...
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
+    <Routes>
+      <Route path="/" element={<Login />} />
+      <Route path="/create-account" element={<CreateUser />} />
+    </Routes>
+  );
```
Comment (ohnomalone) [inline web-ui/src/App.tsx:12]: remove old code and updated with routes to two components

File: web-ui/src/main.tsx [hunk 3]
```diff
@@ -1,10 +1,14 @@
-import React from 'react'
-import ReactDOM from 'react-dom/client'
-import App from './App.tsx'
-import './index.css'
+import React from "react";
+import ReactDOM from "react-dom/client";
+import App from "./App.tsx";
+import "./index.css";
```
Comment (ohnomalone) [inline web-ui/src/main.tsx:4]: I think this is a linter thing, we should sync on what linter we want to use.

File: web-ui/src/main.tsx [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
 
-ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
+import { BrowserRouter } from "react-router-dom";
+
+ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
   <React.StrictMode>
-    <App />
-  </React.StrictMode>,
-)
+    <BrowserRouter>
```
Comment (ohnomalone) [inline web-ui/src/main.tsx:10]: Implement `BrowserRouter` to have routing in the app

File: web-ui/yarn.lock
Comment (ohnomalone) [inline web-ui/yarn.lock:1]: we should look up of this file should be in the .gitignore?

File: web-ui/src/modules/login/login.tsx
Comment (ohnomalone) [inline web-ui/src/modules/login/login.tsx:1]: created component to route to

File: web-ui/src/App.tsx [hunk 5]
```diff
@@ -1,35 +1,15 @@
...
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
+    <Routes>
+      <Route path="/" element={<Login />} />
```
Comment (ohnomalone) [inline web-ui/src/App.tsx:9]: for the route to work, you have to have a path and the element you're linking to

File: web-ui/src/main.tsx (hunk 3 above)
Comment (watermalon3) [reply web-ui/src/main.tsx:4]: I will create a ticket to add Airbnb linter

**Review by watermalon3 — APPROVED** [review]
Amazing work!

**Pull request merged**